qiskit/circuit/quantumcircuit.py

```diff
@@ -2005,10 +2005,12 @@ def assign_parameters(self, param_dict, inplace=False):
         # unroll the parameter dictionary (needed if e.g. it contains a ParameterVector)
         unrolled_param_dict = self._unroll_param_dict(param_dict)
 
-        # check that only existing parameters are in the parameter dictionary
-        if unrolled_param_dict.keys() > self._parameter_table.keys():
+        # check that all param_dict items are in the _parameter_table for this circuit
+        missing_params = [param_key for param_key in unrolled_param_dict
```

Comment:
Could we rename this maybe to `nonexisting_params` or so, to be more clear? When first reading it I thought this would be a list of parameters that are in the circuit but not passed (therefore missing) 😄

Reply:
How about `params_not_in_circuit`? They do exist as Parameters, just not in the circuit (therefore missing??) :-)

Reply:
Works for me 👍🏻